### PR TITLE
survival model chapter with refs

### PR DESCRIPTION
**Pull request by bob-carpenter**
#### Submission Checklist

- [x] Builds locally 
- [x] New functions marked with `` `r since("VERSION")` ``
- [x] Declare copyright holder and open-source license: see below

#### Summary

Add a chapter on parametric and non-parametric survival models with refs.  I've tested that all the code works with Python simulations.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Simons Foundation

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC BY-ND 4.0 (https://creativecommons.org/licenses/by-nd/4.0/)

### CODE REVIEW

File: src/stan-users-guide/survival.Rmd [hunk 1]
```diff
@@ -0,0 +1,788 @@
...
+```
+transformed data {
+  int<lower=0> J = num_unique_starts(t);
+  array[J + 1] int<lower=0> starts = unique_starts(t, J);
+}
+```
+
+For each unique survival time `j` in `1:J`, the subjects
+indexed from `starts[j]` to `starts[j + 1] - 1` (inclusive) share the
+same survival time.  The number of elements with surival time `j` is thus
```
Comment (mitzimorris) [inline src/stan-users-guide/survival.Rmd:745]: ^surival^survival^

File: src/stan-users-guide/survival.Rmd [hunk 2]
```diff
@@ -0,0 +1,788 @@
...
+where $h_0(t)$ is a shared baseline hazard function and $x_n \cdot
+\gamma = \sum_{k=1}^K x_{n, k} \cdot \beta_k$ is a row vector-vector
+product. 
+
+In the semi-parametric, proportional hazards model, the baseline
+hazard function $h_0(t)$ is not modeled.  This is why it is called
+"semi-parametric."  Only the factor $\exp(x_n \cdot \gamma),$ which
+determines how individual $n$ varies by a proportion from the baseline
+hazard, is modeled.  This is why it's called "proportional hazards."
+
```
Comment (mitzimorris) [inline src/stan-users-guide/survival.Rmd:490]: discuss why not generative w/r/t hazard function but is generative w/r/t ordering.

**Review by mitzimorris — CHANGES_REQUESTED** [review]
this is great.
please fix '\[', level 3 headings, and type.
comments are places where a little more exposition might help.

**Comment by bob-carpenter** [comment]
I wasn't entirely clear on what this meant:

> please fix '[', level 3 headings, and type.

I think the first is referring to replacing `\[` with `\begin{equation*}`, which I did.

I think the second is referring to adding `{-}`, which I did.

I couldn't figure out what the third request ("type") meant.

File: src/stan-users-guide/survival.Rmd [hunk 3]
```diff
@@ -0,0 +1,788 @@
+# Survival Models {#survival-models.chapter}
+
+Survival models arise when there is an event of interest for
+a group of subjects that is
```
[the remark this answers is not in the document]
Comment (bob-carpenter) [reply src/stan-users-guide/survival.Rmd:4]: Thanks---I was struggling with this throughout and fixed as requested.

File: src/stan-users-guide/survival.Rmd [hunk 4]
```diff
@@ -0,0 +1,788 @@
...
+\]
+
+The model for the censored failure times is also exponential.  All
+that is known of a censored item is that its failure time is greater
+than the censoring time, so each censored item contributes a factor to
+the likelihood of 
+\[
+\Pr[T > t^{\textrm{cens}}] = 1 - F_T(t^{\textrm{cens}}),
+\]
+where $F_T$ is the cumulative distribution function (cdf) of survival
```
[the remark this answers is not in the document]
Comment (bob-carpenter) [reply src/stan-users-guide/survival.Rmd:90]: F_X is standard notation for the cdf of the random variable X.  I defined it again locally in the doc.

File: src/stan-users-guide/survival.Rmd [hunk 5]
```diff
@@ -488,8 +495,15 @@ hazard function $h_0(t)$ is not modeled.  This is why it is called
 determines how individual $n$ varies by a proportion from the baseline
 hazard, is modeled.  This is why it's called "proportional hazards."
 
+The proportional hazards model is not fully generative.  There is no
+way to generate the times of failure because they for the times of
+failure because the baseline hazard function $h_0(t)$ is unmodeled; if
```
Comment (mitzimorris) [inline src/stan-users-guide/survival.Rmd:500]: "because". .... "because". - something's redundant here

**Review by mitzimorris — APPROVED** [review]
looks great!
one changes needs further editing - then good to go.

**Pull request merged**